include/bluetooth/gatt.h

```diff
@@ -280,7 +286,29 @@ struct bt_gatt_cpf {
  *
  * @return 0 in case of success or negative value in case of error.
  */
-int bt_gatt_register(struct bt_gatt_attr *attrs, size_t count);
+#define bt_gatt_register(_attrs, _count)				\
+	bt_gatt_register_service((&(struct bt_gatt_svc) { .attrs = _attrs, \
+				.attr_count = _count, }))
```

Comment:
How can this work? The anonymous variable you're creating here will be an automatic one, i.e. become invalid as soon as the function that calls bt_gatt_register() returns, e.g. if you call bt_gatt_register() from main() and then return from main() (which is a valid thing to do).